pkg/api/controllers/health/health.go

```diff
@@ -18,6 +20,25 @@ import (
 //	@Router			/health [get]
 //
 //	@id				HealthCheck
+
 func HealthCheck(ctx *gin.Context) {
+	cfg, err := server.GetConfig()
```

Comment:
I see the direction you were going for here but, because all of these components are separated, we shouldn't mix them together here.

The headscale port and local builder registry port can be checked in the server command directly instead of the API here.

Furthermore, the local builder registry might not even be active (because the user disabled it) so you need to take that into account.